test/test_type_promotion.py

```diff
@@ -593,6 +593,38 @@ def test_sparse_true_divide(self, device, dtype):
         casting_result = dividend.to(torch.get_default_dtype()) / 2
         self.assertEqual(casting_result, torch.true_divide(dividend_sparse, 2).to_dense())
 
+    @float_double_default_dtype
+    def test_reductions(self, device):
+        for fn, meth in ((torch.mean, torch.Tensor.mean),
+                         (torch.std, torch.Tensor.std),
+                         (torch.var, torch.Tensor.var),
+                         (torch.logsumexp, torch.Tensor.logsumexp)):
+
+            t = torch.tensor((2, 4), device=device, dtype=torch.int)
+            self.assertTrue(fn(t, dim=0).dtype, torch.get_default_dtype())
```

Comment:
Shouldn't this be assertEqual?

Reply:
Yep! Thanks for pointing that out. Still WIP.